.gitattributes

```diff
@@ -1,7 +1,4 @@
 * text=auto
-*.cs eol=crlf
```

Comment:
done

Reply:
text=auto is supposed to do the conversion automatically. I don't think it's a good idea to force crlf.

Reply:
And each extension can be forced for a specific format (text or binary)

Examples: 
https://github.com/dotnet/dotnet/blob/main/src/aspire/.gitattributes
https://github.com/dotnet/dotnet/blob/main/src/aspnetcore/.gitattributes

Reply:
It's a Roslyn bug: https://github.com/dotnet/roslyn/issues/55526

Reply:
I read the comments, and they don't seem to imply there is a bug, I don't think so either. Sam correctly mentions the end_of_line should not be forced and that's it:

It also needs to remove this from .editorconfig and there won't be conflicting behaviors.

```
end_of_line = crlf
```

Reply:
Mike did the changes, the CI is green. I also cloned on Linux after these changes, did a dotnet format and no files were updated.

And it's LF:

```
00000000: 7573 696e 6720 5379 7374 656d 3b0a 7573  using System;.us
00000010: 696e 6720 5379 7374 656d 2e43 6f6c 6c65  ing System.Colle
00000020: 6374 696f 6e73 2e47 656e 6572 6963 3b0a  ctions.Generic;.
00000030: 7573 696e 6720 5379 7374 656d 2e43 6f6c  using System.Col
00000040: 6c65 6374 696f 6e73 2e49 6d6d 7574 6162  lections.Immutab
```

Reply:
We'll see :).

Reply:
There is a possibility there will be some glitches, but I think only from files that are incorrectly tagged as text vs. binary, or some which are currently cr-lf but should be lf only (web assets for instance).

I can have a look at one of the repositories where you have had the issue if you want and see if I can understand why you are getting problems.

Reply:
It seems to me the recommendation is to enforce the same line ending everywhere and not let the platform handle it: https://github.com/dotnet/roslyn/issues/55526#issuecomment-1165001300 I don't particularly like this, but we're actually doing that in the Lombiq repos (including `end_of_line = crlf` in the .editorconfig) and it works reliably, no issues since then.

But perhaps indeed if we removed all of this then it'd also work.

This was a problem with all of our projects using https://github.com/Lombiq/.NET-Analyzers, i.e. where [these props and config files](https://github.com/Lombiq/.NET-Analyzers/tree/dev/Lombiq.Analyzers) are loaded. Thank you!